packages/experience-builder-sdk/src/hooks/useDetectEditorMode.tsx

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useRef, useState } from 'react';
+import {
+  doesMismatchMessageSchema,
+  sendMessage,
+  tryParseMessage,
+} from '@contentful/experience-builder-core';
+import { INCOMING_EVENTS, OUTGOING_EVENTS } from '@contentful/experience-builder-core/constants';
+
+export const useDetectEditorMode = () => {
+  //Assume we are in editor mode initially if we are in iframe
+  const [isEditorMode, setIsEditorMode] = useState(inIframe());
+  const receivedMessage = useRef(false);
+
+  useEffect(() => {
+    const onMessage = (event: MessageEvent) => {
+      if (doesMismatchMessageSchema(event)) {
+        return;
+      }
+      const eventData = tryParseMessage(event);
+
+      if (eventData.eventType === INCOMING_EVENTS.RequestEditorMode) {
+        setIsEditorMode(true);
+        receivedMessage.current = true;
+        if (typeof window !== 'undefined') {
+          //Once we definitely know that we are in editor mode, we set this flag so future postMessage connect calls are not made
+          window.__EB__.isEditorMode = true;
```

Comment:
since the word `builder` is removed from the product name, should we pick a different alias instead of `EB`?

Reply:
we aren't making a concerted effort to rename EB in the codebase at this time, so I think this is fine.